libs/com.etendorx.das_core/src/main/java/com/etendorx/entities/mapper/lib/JsonPathConverterBase.java

```diff
@@ -18,13 +18,84 @@
 import com.jayway.jsonpath.Configuration;
 import com.jayway.jsonpath.DocumentContext;
 import com.jayway.jsonpath.JsonPath;
-import com.jayway.jsonpath.Option;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This abstract class provides a base implementation for a JsonPathConverter.
+ * It provides methods for reading JSON data using JsonPath and validating the read values.
+ *
+ * @param <E> The type of the object that will be returned after conversion.
+ */
+@Log4j2
 public abstract class JsonPathConverterBase<E> implements JsonPathConverter<E> {
+
+  /**
+   * Reads a value from the provided DocumentContext at the specified path and returns it as an instance of the provided class.
+   * If the value cannot be read or is not of the expected class, it returns a ReturnKey with error information.
+   *
+   * @param ctx The DocumentContext from which to read the value.
+   * @param path The JsonPath at which to read the value.
+   * @param clazz The class of the value to be read.
+   * @return A ReturnKey containing the read value and error information.
+   */
+  public <E> ReturnKey<E> read(DocumentContext ctx, String path, Class<E> clazz) {
```

Comment:
public <E> ReturnKey<E> read(DocumentContext ctx, String path, Class<E> clazz) {
    try {
      E value = ctx.read(path, clazz);
      log.debug("    readedPath '{}' '{}'", path, value);
      // Is an error if the value is not of the expected class
      var nullValue = (!clazz.isAssignableFrom(value.getClass());
      return new ReturnKey<>(path, value, false, nullValue, nullValue);
    } catch (Exception e) {
      log.debug("    Cannot read path '{}'", path);
      // Is not considered an error if the path is not found
      return new ReturnKey<>(path, null, true, true, false);
    }
  }